packages/mattermost-redux/src/actions/users.ts

```diff
@@ -104,113 +100,25 @@ export function loginById(id: string, password: string, mfaToken = ''): ActionFu
         dispatch({type: UserTypes.LOGIN_REQUEST, data: null});
 
         const deviceId = getState().entities.general.deviceToken;
-        let data;
 
         try {
-            data = await Client4.loginById(id, password, mfaToken, deviceId);
+            await Client4.loginById(id, password, mfaToken, deviceId);
         } catch (error) {
             dispatch(batchActions([
                 {
                     type: UserTypes.LOGIN_FAILURE,
                     error,
                 },
-                logError(error),
-            ]));
-            return {error};
-        }
-
-        return completeLogin(data)(dispatch, getState);
-    };
-}
-
-function completeLogin(data: UserProfile): ActionFunc {
-    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
```

Comment:
Nice find! Even knowing what to check, it took a lot of jumping around between the two and the things they call to see that it is all doing the same thing.